### ISSUE TITLE
Crash report from Dynamo 2.18.0.4931

### ISSUE THREAD

**Issue by angelowang** [DynamoDS/Dynamo#14112]
# Issue Description
- Launch DynamoSandbox.exe
- Create a python node
- Double-click to edit the script
- Before the last OUT=0 line, paste the following code:
```
with adoc.LockDocument():
    with adoc.Database as db:

        with db.TransactionManager.StartTransaction() as t:
            # Place your code below
            # 
            #

            # Commit before end transaction
            #t.Commit()
            pass
```
- Crash dialog appears.
- Civil 3D is using the python template with the above code, which means
    - When user double click a newly created python node, we will get an exception dialog
    - User will never be able to edit the script after creating the node.

## Dynamo version
Dynamo: `2.18.0.4931`

## Operating system
OS: `Microsoft Windows NT 6.2.9200.0`

## What did you do?
Autodesk Dynamo for Civil 3D

## What did you expect to see?
No crash.

## What did you see instead?
Crash, and this leads to further crash on GC in AutoCAD.

## What packages or external references (if any) were used?
No loaded packages were found.

## Stack Trace
```
(From the Dynamo crash window select 'Details' -> 'Copy' and paste here)
Stack empty.

   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at System.Collections.Generic.Stack`1.Pop()
   at PythonNodeModelsWpf.TabFoldingStrategy.CreateNewFoldingsByLine(ITextSource document)
   at PythonNodeModelsWpf.ScriptEditorWindow.UpdateFoldings()
   at ICSharpCode.AvalonEdit.TextEditor.OnTextChanged(EventArgs e)
   at ICSharpCode.AvalonEdit.TextEditor.ReceiveWeakEvent(Type managerType, Object sender, EventArgs e)
   at System.Windows.WeakEventManager.ListenerList.DeliverEvent(Listener& listener, Object sender, EventArgs args, Type managerType)
   at System.Windows.WeakEventManager.ListenerList.DeliverEvent(Object sender, EventArgs args, Type managerType)
   at System.Windows.WeakEventManager.DeliverEvent(Object sender, EventArgs args)
   at ICSharpCode.AvalonEdit.Document.TextDocument.FireChangeEvents()
   at ICSharpCode.AvalonEdit.Document.TextDocument.EndUpdate()
   at ICSharpCode.AvalonEdit.Document.TextDocument.Replace(Int32 offset, Int32 length, String text, OffsetChangeMap offsetChangeMap)
   at ICSharpCode.AvalonEdit.Editing.EmptySelection.ReplaceSelectionWithText(String newText)
   at ICSharpCode.AvalonEdit.Editing.EditingCommandHandler.OnPaste(Object target, ExecutedRoutedEventArgs args)
   at System.Windows.Input.CommandBinding.OnExecuted(Object sender, ExecutedRoutedEventArgs e)
   at System.Windows.Input.CommandManager.ExecuteCommandBinding(Object sender, ExecutedRoutedEventArgs e, CommandBinding commandBinding)
   at System.Windows.Input.CommandManager.FindCommandBinding(CommandBindingCollection commandBindings, Object sender, RoutedEventArgs e, ICommand command, Boolean execute)
   at System.Windows.Input.CommandManager.FindCommandBinding(Object sender, RoutedEventArgs e, ICommand command, Boolean execute)
   at System.Windows.Input.CommandManager.OnExecuted(Object sender, ExecutedRoutedEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.RoutedCommand.ExecuteImpl(Object parameter, IInputElement target, Boolean userInitiated)
   at System.Windows.Input.CommandManager.TranslateInput(IInputElement targetElement, InputEventArgs inputEventArgs)
   at System.Windows.UIElement.OnKeyDownThunk(Object sender, KeyEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndKeyboardInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawKeyboardActions actions, Int32 scanCode, Boolean isExtendedKey, Boolean isSystemKey, Int32 virtualKey)
   at System.Windows.Interop.HwndKeyboardInputProvider.ProcessKeyAction(MSG& msg, Boolean& handled)
   at System.Windows.Interop.HwndSource.CriticalTranslateAccelerator(MSG& msg, ModifierKeys modifiers)
   at System.Windows.Interop.HwndSource.OnPreprocessMessage(Object param)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at System.Windows.Threading.Dispatcher.Invoke(DispatcherPriority priority, Delegate method, Object arg)
   at System.Windows.Interop.HwndSource.OnPreprocessMessageThunk(MSG& msg, Boolean& handled)
   at System.Windows.Interop.ThreadMessageEventHandler.Invoke(MSG& msg, Boolean& handled)
   at System.Windows.Interop.ComponentDispatcherThread.RaiseThreadMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at DynamoSandbox.DynamoCoreSetup.RunApplication(Application app)
```

---
CLR: `4.0.30319.42000`

**Comment by QilongTang** [comment]
I cant reproduce using the latest sandbox. @avidit Can you try with previous releases?

**Comment by mjkkirschner** [comment]
@QilongTang I think this was likely fixed here:
https://github.com/DynamoDS/Dynamo/pull/14004/files
which should be in 2.18.1 - @angelowang can you try that hotfix version?

**Comment by QilongTang** [comment]
Thank you @mjkkirschner same comment here, I thought C3D integrated 2.18.1..

**Comment by angelowang** [comment]
@QilongTang @mjkkirschner Good to see this. This is indeed fixed in 2.18.1. I will close this and work with C3D team on the next plans.
